File: 1009-complement-of-base-10-integer/1009-complement-of-base-10-integer.java
class Solution {
    public int bitwiseComplement(int n) {
        int ans = 0,temp = 1;
        if(n == 0) return 1;
        while(n > 0)
        {
            if((n & 1) == 0)
            {
                ans += temp;
            }
            temp *= 2;
            n >>= 1;
        }
        return ans;
    }
}

File: 1009-complement-of-base-10-integer/1009-complement-of-base-10-integer.cpp
class Solution {
public:
    int bitwiseComplement(int n) {
        if(n==0) return 1;
        string binary="";
        while(n)
        {
            binary=(to_string(!(n&1)))+binary;
            n>>=1;
        }
        // cout<<binary;
        return stoi(binary,0,2);
    }
};